examples/with_configuration/pyproject.toml

```diff
@@ -0,0 +1,13 @@
+[tool.rstcheck]
+ignore_directives = [
+	"foobar",
+	"my-directive"
+]
+ignore_roles = [
+	"some-custom-thing"
+]
+ignore_messages = [
+	"(Document or section may not begin with a transition\\.$)"
+]
+ignore_language = "cpp"
```

Comment:
I do not like the idea to use comma-separated-string even so we have a list type in toml at hand.
When `ignore_language` is a string, why are the other ignore_* not?

Reply:
I only made this change to the example pyproject.toml to test both list and string for the ignore_* options.

See also my comment above, but, comma-separated-strings are allowable in pyproject.toml unless specifically prohibited by the tool from what I've seen.  For example, black does the following to constrain the target-version option to a list:

``` 
target_version = config.get("target_version")
if target_version is not None and not isinstance(target_version, list):
    raise click.BadOptionUsage(
        "target-version", "Config key target-version must be a list"
    )
```

Without code to constrain pyproject.toml options to lists, the savvy user **might** use comma-separated-strings.  But the documentation for pyproject.toml implies lists are required.  There is no discussion in the documentation about using comma-separated-strings as far as I know.  So I think common practice would be either lists, simple strings, or integers.

I can add code to constrain options to lists, but that's going to take me longer especially this time of year (spring is planting season).  But, it's starting to feel to me like all the config file code needs to be restructured (like different classes for different config file formats) which would seem to fit with your plan to split things up after v6.0.0 is released.

Anyway, I'm not trying to be difficult, I've just been conditioned that way after years in the nuclear and aerospace industries ;).  Just let me know how you want to proceed.